From: Mike Hommey <mh@glandium.org>
Date: Sat, 6 Feb 2021 07:52:19 +0900
Subject: Allow to build with cargo in Debian unstable

---
 build/moz.configure/rust.configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index ff3dbe0..2ceff72 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -168,7 +168,7 @@ def rust_compiler(rustc_info, cargo_info, build_project):
         rustc_min_version = Version("1.47.0")
     else:
         rustc_min_version = Version(MINIMUM_RUST_VERSION)
-    cargo_min_version = rustc_min_version
+    cargo_min_version = Version("1.46.0")
 
     version = rustc_info.version
     is_nightly = "nightly" in version.version
