From 84a08f543ec910d3fc0da4582cd79ba4a8fd22d9 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Wed, 26 Apr 2017 21:41:24 +0200
Subject: [PATCH 1/5] fixed default profile name

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 toolkit/xre/nsAppRunner.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index cdec7dfa8418..19cdcc1b1731 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -3832,6 +3832,14 @@ XREMain::XRE_mainStartup(bool* aExitFlag)
       }
     }
 
+    if (!profile) {
+#ifdef MOZ_DEV_EDITION
+      profile = "dev-edition-default";
+#else
+      profile = "default";
+#endif
+    }
+
     nsCOMPtr<nsIFile> mutexDir;
     rv = GetSpecialSystemDirectory(OS_TemporaryDirectory, getter_AddRefs(mutexDir));
     if (NS_SUCCEEDED(rv)) {
-- 
2.11.0

