From 4e26ce54f5f78a13663f448853a6f1033410ca1a Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Thu, 25 May 2017 17:55:33 +0900
Subject: [PATCH] Fix setting up shellHasCSD flag position

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/nsWindow.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index bf68d7604dae..b881a876bf75 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -3799,10 +3799,6 @@ nsWindow::Create(nsIWidget* aParent,
         // Create a container to hold child windows and child GtkWidgets.
         GtkWidget *container = moz_container_new();
         mContainer = MOZ_CONTAINER(container);
-        // We use mContainer to draw on Wayland
-        if (!mIsX11Display) {
-            shellHasCSD = true;
-        }
 #if (MOZ_WIDGET_GTK == 3)
         // "csd" style is set when widget is realized so we need to call
         // it explicitly now.
@@ -3814,6 +3810,10 @@ nsWindow::Create(nsIWidget* aParent,
 
         // Always draw to mozcontainer on Wayland
         shellHasCSD = gtk_style_context_has_class(style, "csd");
+        // We use mContainer to draw on Wayland
+        if (!mIsX11Display) {
+            shellHasCSD = true;
+        }
 #endif
         if (!shellHasCSD) {
             // Use mShell's window for drawing and events.
-- 
2.11.0

