From 622e8fd2f042c0d66e86444a1a5a4bf9cfcb08aa Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Tue, 25 Apr 2017 16:48:24 +0900
Subject: [PATCH] Add a missing CXXFLAGS for dbus

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 toolkit/components/remote/moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/toolkit/components/remote/moz.build b/toolkit/components/remote/moz.build
index 50b13b768fe1..b3601f0d1341 100644
--- a/toolkit/components/remote/moz.build
+++ b/toolkit/components/remote/moz.build
@@ -21,3 +21,6 @@ if 'gtk' in CONFIG['MOZ_WIDGET_TOOLKIT']:
 FINAL_LIBRARY = 'xul'
 
 CXXFLAGS += CONFIG['TK_CFLAGS']
+
+if CONFIG['ENABLE_REMOTE_DBUS']:
+    CXXFLAGS += CONFIG['MOZ_DBUS_GLIB_CFLAGS']
-- 
2.11.0

