From ee5396c268ffa1d2360bcc96fc6898b398033ea2 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Wed, 10 May 2017 11:28:36 +0200
Subject: [PATCH] Call wl_display_roundtrip() twice to ensure we have valid
 data offer

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/nsClipboardWayland.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/widget/gtk/nsClipboardWayland.cpp b/widget/gtk/nsClipboardWayland.cpp
index a7e8a215eb96..46c033363f25 100644
--- a/widget/gtk/nsClipboardWayland.cpp
+++ b/widget/gtk/nsClipboardWayland.cpp
@@ -319,6 +319,7 @@ nsRetrievalContextWayland::nsRetrievalContextWayland(void)
     wl_registry_add_listener(wl_display_get_registry(mDisplay),
                              &clipboard_registry_listener, this);
     wl_display_roundtrip(mDisplay);
+    wl_display_roundtrip(mDisplay);
 
     // We don't have Wayland support here so just give up
     if (!mDataDeviceManager || !mSeat)
@@ -327,6 +328,10 @@ nsRetrievalContextWayland::nsRetrievalContextWayland(void)
     wl_data_device *dataDevice =
         wl_data_device_manager_get_data_device(mDataDeviceManager, mSeat);
     wl_data_device_add_listener(dataDevice, &data_device_listener, this);
+    // We have to call wl_display_roundtrip() twice otherwise data_offer_listener
+    // may not be processed because it's called from data_device_data_offer
+    // callback.
+    wl_display_roundtrip(mDisplay);
     wl_display_roundtrip(mDisplay);
 
     mInitialized = true;
-- 
2.11.0

