From 7b8cf7f93879cf2a1f20332a4f57fa9b6a306121 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Fri, 26 May 2017 15:42:29 +0200
Subject: [PATCH] Don't crash when we're missing clipboard data, rhbz#1455915

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/nsClipboardWayland.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/widget/gtk/nsClipboardWayland.cpp b/widget/gtk/nsClipboardWayland.cpp
index 46c033363f25..30bec8438751 100644
--- a/widget/gtk/nsClipboardWayland.cpp
+++ b/widget/gtk/nsClipboardWayland.cpp
@@ -384,6 +384,14 @@ nsRetrievalContextWayland::GetClipboardContent(const char* aMimeType,
 {
     NS_ASSERTION(mDataOffer, "Requested data without valid data offer!");
 
+    if (!mDataOffer) {
+        // TODO
+        // Something went wrong. We're requested to provide clipboard data
+        // but we haven't got any from wayland. Looks like rhbz#1455915.
+        // Return NS_ERROR_FAILURE to avoid crash.
+        return NS_ERROR_FAILURE;
+    }
+
     int pipe_fd[2];
     if (pipe(pipe_fd) == -1)
         return NS_ERROR_FAILURE;
-- 
2.11.0

