From 73605ca41ed93b90507733ad28ff2ae2724704de Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Thu, 20 Apr 2017 09:43:48 +0900
Subject: [PATCH] Adopt wl_data_offer_listener for older Wayland under 1.9

---
 widget/gtk/nsClipboardWayland.cpp | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/widget/gtk/nsClipboardWayland.cpp b/widget/gtk/nsClipboardWayland.cpp
index a7e8a215eb96..6962a80d8850 100644
--- a/widget/gtk/nsClipboardWayland.cpp
+++ b/widget/gtk/nsClipboardWayland.cpp
@@ -112,24 +112,8 @@ data_offer_offer (void                 *data,
   context->AddMIMEType(type);
 }
 
-static void
-data_offer_source_actions(void *data,
-                          struct wl_data_offer *wl_data_offer,
-                          uint32_t source_actions)
-{
-}
-
-static void
-data_offer_action(void *data,
-                  struct wl_data_offer *wl_data_offer,
-                  uint32_t dnd_action)
-{
-}
-
 static const struct wl_data_offer_listener data_offer_listener = {
     data_offer_offer,
-    data_offer_source_actions,
-    data_offer_action
 };
 
 static void
-- 
2.11.0

