From 2403a91dd4e2da64f784d6764427467797380d35 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Wed, 12 Jul 2017 12:02:25 +0200
Subject: [PATCH 8/8] Remove unrealize handler (rhbz#1467104)

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/mozcontainer.c | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/widget/gtk/mozcontainer.c b/widget/gtk/mozcontainer.c
index 420329e84956..3523abd481c0 100644
--- a/widget/gtk/mozcontainer.c
+++ b/widget/gtk/mozcontainer.c
@@ -26,9 +26,6 @@ static void moz_container_init                (MozContainer      *container);
 static void moz_container_map                 (GtkWidget         *widget);
 static void moz_container_unmap               (GtkWidget         *widget);
 static void moz_container_realize             (GtkWidget         *widget);
-#if defined(MOZ_WAYLAND)
-static void moz_container_unrealize           (GtkWidget         *widget);
-#endif
 static void moz_container_size_allocate       (GtkWidget         *widget,
                                                GtkAllocation     *allocation);
 
@@ -214,9 +211,6 @@ moz_container_class_init (MozContainerClass *klass)
     widget_class->map = moz_container_map;
     widget_class->unmap = moz_container_unmap;
     widget_class->realize = moz_container_realize;
-#if defined(MOZ_WAYLAND)
-    widget_class->unrealize = moz_container_unrealize;
-#endif
     widget_class->size_allocate = moz_container_size_allocate;
 
     container_class->remove = moz_container_remove;
@@ -384,23 +378,6 @@ moz_container_realize (GtkWidget *widget)
 #endif
 }
 
-#if defined(MOZ_WAYLAND)
-static void
-moz_container_unrealize (GtkWidget *widget)
-{
-  MozContainer* container = MOZ_CONTAINER(widget);
-  moz_container_unmap_surface(container);
-
-  gtk_widget_set_realized(widget, FALSE);
-
-  GdkWindow* window = gtk_widget_get_window(widget);
-  gdk_window_set_user_data(window, nullptr);
-  gdk_window_destroy(window);
-
-  gtk_widget_set_window(widget, nullptr);
-}
-#endif
-
 void
 moz_container_size_allocate (GtkWidget     *widget,
                              GtkAllocation *allocation)
-- 
2.11.0

