From 2606b093727fb189281988150327655e1fd61e15 Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Tue, 25 Apr 2017 14:43:28 +0900
Subject: [PATCH 9/9] Build fixes for destructor argument mismatch

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 toolkit/components/remote/nsGTKRemoteService.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/components/remote/nsGTKRemoteService.h b/toolkit/components/remote/nsGTKRemoteService.h
index 638558b4398a..ae18de7a9cdc 100644
--- a/toolkit/components/remote/nsGTKRemoteService.h
+++ b/toolkit/components/remote/nsGTKRemoteService.h
@@ -40,7 +40,7 @@ public:
     { }
 
     DBusHandlerResult HandleDBusMessage(DBusConnection *aConnection, DBusMessage *msg);
-    void UnregisterDBusInterface();
+    void UnregisterDBusInterface(DBusConnection *aConnection);
 
 private:
   ~nsGTKRemoteService() { }
-- 
2.11.0

