From d8dc4298fa3ce9210aa5be4322c9b1c8540ea581 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Mon, 27 Mar 2017 10:52:23 +0200
Subject: [PATCH 14/44] Removed wayland-client from libxul.so

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 toolkit/library/moz.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/toolkit/library/moz.build b/toolkit/library/moz.build
index a7724c4130d6..a9bdf123dd32 100644
--- a/toolkit/library/moz.build
+++ b/toolkit/library/moz.build
@@ -313,9 +313,6 @@ if 'gtk' in CONFIG['MOZ_WIDGET_TOOLKIT']:
     OS_LIBS += [
         'gthread-2.0',
     ]
-    OS_LIBS += [
-        'wayland-client',
-    ]
 
 if CONFIG['MOZ_ENABLE_STARTUP_NOTIFICATION']:
     OS_LIBS += CONFIG['MOZ_STARTUP_NOTIFICATION_LIBS']
-- 
2.11.0

