From 8d8973f1a1c8e8c59aee39fdf5b4036ffd04bc85 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Tue, 11 Apr 2017 11:56:43 +0200
Subject: [PATCH 29/44] Don't read wayland events when poll fails

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/WindowSurfaceWayland.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/widget/gtk/WindowSurfaceWayland.cpp b/widget/gtk/WindowSurfaceWayland.cpp
index 3d91a3f53f4b..8b0ec3e4c3c6 100644
--- a/widget/gtk/WindowSurfaceWayland.cpp
+++ b/widget/gtk/WindowSurfaceWayland.cpp
@@ -124,9 +124,10 @@ WaylandDisplayLoop(gpointer data)
       int ret = poll(&fds, 1, -1);
       if (ret == -1) {
           wl_display_cancel_read(display);
+      } else {
+          wl_display_read_events(display);
+          wl_display_dispatch_queue_pending(display, eventQueue);
       }
-      wl_display_read_events(display);
-      wl_display_dispatch_queue_pending(display, eventQueue);
   }
 
   return nullptr;
-- 
2.11.0

