From 67c10d84e29514306c773c2cc9a4f67325236434 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@homa.ne.jp>
Date: Tue, 6 Sep 2016 14:17:51 +0900
Subject: [PATCH 40/44] GLContextPrividerEGL: Remove needless code

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 gfx/gl/GLContextProviderEGL.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gfx/gl/GLContextProviderEGL.cpp b/gfx/gl/GLContextProviderEGL.cpp
index ad478e0bd2bb..bd8369b35a11 100644
--- a/gfx/gl/GLContextProviderEGL.cpp
+++ b/gfx/gl/GLContextProviderEGL.cpp
@@ -4,10 +4,10 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #if defined(MOZ_WIDGET_GTK)
-    #include <gdk/gdkx.h>
     #if defined(MOZ_WAYLAND)
       #define GET_NATIVE_WINDOW(aWidget) ((EGLNativeWindowType)aWidget->GetNativeData(NS_NATIVE_EGL_WINDOW))
     #else
+      #include <gdk/gdkx.h>
       // we're using default display for now
       #define GET_NATIVE_WINDOW(aWidget) ((EGLNativeWindowType)GDK_WINDOW_XID((GdkWindow*)aWidget->GetNativeData(NS_NATIVE_WINDOW)))
    #endif
-- 
2.11.0

