From 4f55c09b4bf6fb00f6ffd99b8abd5372553fdabf Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@homa.ne.jp>
Date: Wed, 28 Sep 2016 14:36:02 +0900
Subject: [PATCH 43/44] Add --with-gl-provider=EGL to .mozconfig

Signed-off-by: Takuro Ashie <ashie@homa.ne.jp>
Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 .mozconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.mozconfig b/.mozconfig
index 0389b40aaf88..dd4c44965baf 100644
--- a/.mozconfig
+++ b/.mozconfig
@@ -1,6 +1,7 @@
 . $topsrcdir/browser/config/mozconfig
 
 ac_add_options --enable-default-toolkit=cairo-gtk3-wayland
+ac_add_options --with-gl-provider=EGL
 
 mk_add_options BUILD_OFFICIAL=1
 mk_add_options MOZILLA_OFFICIAL=1
-- 
2.11.0

