diff -Nur firefox-51.0b9-bak/dom/bindings/GenerateCSS2PropertiesWebIDL.py firefox-51.0b9/dom/bindings/GenerateCSS2PropertiesWebIDL.py
--- firefox-51.0b9-bak/dom/bindings/GenerateCSS2PropertiesWebIDL.py	2016-11-15 01:12:27.000000000 +0900
+++ firefox-51.0b9/dom/bindings/GenerateCSS2PropertiesWebIDL.py	2016-12-26 17:17:30.321804897 +0900
@@ -16,7 +16,7 @@
     return "  [%s] attribute DOMString %s;\n" % (", ".join(extendedAttrs),
                                                  propName)
 def generate(output, idlFilename, preprocessorHeader):
-    cpp = list(buildconfig.substs['CPP'])
+    cpp = shellutil.split(buildconfig.substs['CPP'])
     cpp += shellutil.split(buildconfig.substs['ACDEFINES'])
     cpp.append(preprocessorHeader)
     preprocessed = subprocess.check_output(cpp)
diff -Nur firefox-51.0b9-bak/layout/style/GenerateCSSPropsGenerated.py firefox-51.0b9/layout/style/GenerateCSSPropsGenerated.py
--- firefox-51.0b9-bak/layout/style/GenerateCSSPropsGenerated.py	2016-11-15 01:12:29.000000000 +0900
+++ firefox-51.0b9/layout/style/GenerateCSSPropsGenerated.py	2016-12-26 17:17:58.445804622 +0900
@@ -10,7 +10,7 @@
 from mozbuild import shellutil
 
 def get_properties(preprocessorHeader):
-    cpp = list(buildconfig.substs['CPP'])
+    cpp = shellutil.split(buildconfig.substs['CPP'])
     cpp += shellutil.split(buildconfig.substs['ACDEFINES'])
     cpp.append(preprocessorHeader)
     preprocessed = subprocess.check_output(cpp)
