diff -Nur firefox-51.0b5-bak/configure.py firefox-51.0b5/configure.py
--- firefox-51.0b5-bak/configure.py	2016-11-15 01:12:24.000000000 +0900
+++ firefox-51.0b5/configure.py	2016-12-01 13:54:48.282506802 +0900
@@ -68,7 +68,7 @@
             from __future__ import unicode_literals
             from mozbuild.util import encode
             encoding = '%(encoding)s'
-        ''') % {'python': config['PYTHON'], 'encoding': encoding})
+        ''') % {'python': '/usr/bin/python', 'encoding': encoding})
         # A lot of the build backend code is currently expecting byte
         # strings and breaks in subtle ways with unicode strings. (bug 1296508)
         for k, v in sanitized_config.iteritems():
