diff -Nur firefox-52.0b1-bak/gfx/skia/skia/src/core/SkBitmapProcState.cpp firefox-52.0b1/gfx/skia/skia/src/core/SkBitmapProcState.cpp
--- firefox-52.0b1-bak/gfx/skia/skia/src/core/SkBitmapProcState.cpp	2017-01-24 01:13:43.000000000 +0900
+++ firefox-52.0b1/gfx/skia/skia/src/core/SkBitmapProcState.cpp	2017-01-27 14:49:42.926930916 +0900
@@ -300,7 +300,7 @@
                 return false;
         }
 
-#if !defined(SK_ARM_HAS_NEON)
+#if !defined(SK_ARM_HAS_NEON) || defined(SK_ARM_HAS_OPTIONAL_NEON)
         static const SampleProc32 gSkBitmapProcStateSample32[] = {
             S32_opaque_D32_nofilter_DXDY,
             S32_alpha_D32_nofilter_DXDY,
diff -Nur firefox-52.0b1-bak/gfx/skia/skia/src/core/SkBitmapProcState_matrixProcs.cpp firefox-52.0b1/gfx/skia/skia/src/core/SkBitmapProcState_matrixProcs.cpp
--- firefox-52.0b1-bak/gfx/skia/skia/src/core/SkBitmapProcState_matrixProcs.cpp	2017-01-24 01:13:43.000000000 +0900
+++ firefox-52.0b1/gfx/skia/skia/src/core/SkBitmapProcState_matrixProcs.cpp	2017-01-27 14:49:42.926930916 +0900
@@ -56,7 +56,7 @@
 #endif // defined(SK_ARM_HAS_NEON)
 
 // Compile non-neon code path if needed
-#if !defined(SK_ARM_HAS_NEON)
+#if !defined(SK_ARM_HAS_NEON) || defined(SK_ARM_HAS_OPTIONAL_NEON)
 #define MAKENAME(suffix)        ClampX_ClampY ## suffix
 #define TILEX_PROCF(fx, max)    SkClampMax((fx) >> 16, max)
 #define TILEY_PROCF(fy, max)    SkClampMax((fy) >> 16, max)
