diff -Nur firefox-48.0-bak/configure.py firefox-48.0/configure.py
--- firefox-48.0-bak/configure.py	2016-08-17 17:06:29.349136156 +0900
+++ firefox-48.0/configure.py	2016-08-17 17:15:03.073156160 +0900
@@ -57,7 +57,7 @@
     print("Creating config.status", file=sys.stderr)
     encoding = 'mbcs' if sys.platform == 'win32' else 'utf-8'
     with codecs.open('config.status', 'w', encoding) as fh:
-        fh.write('#!%s\n' % config['PYTHON'])
+        fh.write('#!%s\n' % "/usr/bin/python")
         fh.write('# coding=%s\n' % encoding)
         # Because we're serializing as JSON but reading as python, the values
         # for True, False and None are true, false and null, which don't exist.
