diff --git a/python/mozbuild/mozbuild/backend/recursivemake.py b/python/mozbuild/mozbuild/backend/recursivemake.py
index 5192d45..8e769ef 100644
--- a/python/mozbuild/mozbuild/backend/recursivemake.py
+++ b/python/mozbuild/mozbuild/backend/recursivemake.py
@@ -486,8 +486,15 @@ class RecursiveMakeBackend(CommonBackend):
             else:
                 base = backend_file.srcdir
             for f in sorted(obj.files):
+                _variables = variables
+                if obj.canonical_suffix == '.S':
+                    extMatch = re.compile('^(.*?(?:_asm)?)(?:_asm)*?\.asm\.S$').search(f)
+                    if extMatch:
+                        f = extMatch.group(1) + '.asm'
+                        _variables = map(lambda entry: suffix_map['.s'] if entry==suffix_map['.S'] else entry, variables)
+
                 f = mozpath.relpath(f, base)
-                for var in variables:
+                for var in _variables:
                     backend_file.write('%s += %s\n' % (var, f))
         elif isinstance(obj, HostSources):
             suffix_map = {
diff --git a/config/rules.mk b/config/rules.mk
index ed56ca7..88fdeb6 100644
--- a/config/rules.mk
+++ b/config/rules.mk
@@ -209,7 +209,7 @@ CPPOBJS = $(notdir $(addsuffix .$(OBJ_SUFFIX),$(basename $(CPPSRCS))))
 CMOBJS = $(notdir $(CMSRCS:.m=.$(OBJ_SUFFIX)))
 CMMOBJS = $(notdir $(CMMSRCS:.mm=.$(OBJ_SUFFIX)))
 # ASFILES can have different extensions (.s, .asm)
-ASOBJS = $(notdir $(addsuffix .$(OBJ_SUFFIX),$(basename $(ASFILES))))
+ASOBJS = $(notdir $($(ASFILES:.asm=.$(OBJ_SUFFIX)):.s=$(OBJ_SUFFIX)))
 RS_STATICLIB_CRATE_OBJ = $(addprefix lib,$(notdir $(RS_STATICLIB_CRATE_SRC:.rs=.$(LIB_SUFFIX))))
 ifndef OBJS
 _OBJS = $(COBJS) $(SOBJS) $(CPPOBJS) $(CMOBJS) $(CMMOBJS) $(ASOBJS)
