diff -Nur firefox-50.0-bak/gfx/thebes/gfxPlatformGtk.cpp firefox-50.0/gfx/thebes/gfxPlatformGtk.cpp
--- firefox-50.0-bak/gfx/thebes/gfxPlatformGtk.cpp	2016-11-01 05:15:32.000000000 +0900
+++ firefox-50.0/gfx/thebes/gfxPlatformGtk.cpp	2016-11-17 12:02:42.579756063 +0900
@@ -891,11 +891,11 @@
   return gfxPlatform::CreateHardwareVsyncSource();
 }
 
+#endif
+
 bool
 gfxPlatformGtk::SupportsApzTouchInput() const
 {
   int value = gfxPrefs::TouchEventsEnabled();
   return value == 1 || value == 2;
 }
-
-#endif
