diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index ba2b5c5abf62..9236de00f993 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -159,6 +159,7 @@ def rust_triple_alias(host_or_target):
     @imports(_from='textwrap', _import='dedent')
     def rust_target(rustc, host_or_target, building_with_gcc,
                     rust_supported_targets):
+        return "armv7-unknown-linux-gnueabihf"
         # Rust's --target options are similar to, but not exactly the same
         # as, the autoconf-derived targets we use.  An example would be that
         # Rust uses distinct target triples for targetting the GNU C++ ABI
