From 32a22b79742b34401ecb2de488bbe72d27276719 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Thu, 26 Oct 2017 16:49:03 +0900
Subject: [PATCH] Wayland: Fix a crash on starting video playback of WebRTC

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 widget/gtk/nsWindow.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index bf3719d9611a..7ff00c038d40 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -5311,7 +5311,10 @@ nsWindow::HideWindowChrome(bool aShouldHide)
     // error later when this happens (when the persistence timer fires
     // and GetWindowPos is called)
 #ifdef MOZ_X11
-    XSync(GDK_DISPLAY_XDISPLAY(gdk_display_get_default()) , False);
+    if (mIsX11Display)
+        XSync(GDK_DISPLAY_XDISPLAY(gdk_display_get_default()) , False);
+    else
+        gdk_flush ();
 #else
     gdk_flush ();
 #endif /* MOZ_X11 */
@@ -7201,6 +7204,8 @@ nsWindow::GetWaylandDisplay()
 wl_surface*
 nsWindow::GetWaylandSurface()
 {
-  return moz_container_get_wl_surface(MOZ_CONTAINER(mContainer));
+  if (IS_MOZ_CONTAINER(mContainer))
+    return moz_container_get_wl_surface(MOZ_CONTAINER(mContainer));
+  return nullptr;
 }
 #endif
-- 
2.11.0

