From a459fbc8295553e7558b7b25c66a48c619eec426 Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Mon, 16 Oct 2017 17:40:05 +0900
Subject: [PATCH] Remove unused show_shell static function

---
 widget/gtk/nsWindow.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index 14d25c6760cd..bf3719d9611a 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -4312,12 +4312,6 @@ nsWindow::NativeMoveResize()
     }
 }
 
-static void
-show_shell(GdkSeat *seat, GdkWindow *window, gpointer user_data)
-{
-    gtk_widget_show(GTK_WIDGET(user_data));
-}
-
 void
 nsWindow::NativeShow(bool aAction)
 {
-- 
2.11.0

