From 6084d417b17268716b1e76c67239f3ee84699dd1 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 25 Apr 2017 18:50:42 +0900
Subject: [PATCH] Add a missing CXXFLAGS for nsGTKRemoteService

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 toolkit/components/remote/moz.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/toolkit/components/remote/moz.build b/toolkit/components/remote/moz.build
index 50b13b7..35b42c3 100644
--- a/toolkit/components/remote/moz.build
+++ b/toolkit/components/remote/moz.build
@@ -14,6 +14,8 @@ SOURCES += [
     'nsXRemoteService.cpp',
 ]
 if 'gtk' in CONFIG['MOZ_WIDGET_TOOLKIT']:
+    if CONFIG['MOZ_ENABLE_DBUS']:
+        CXXFLAGS += CONFIG['MOZ_DBUS_GLIB_CFLAGS']
     SOURCES += [
         'nsGTKRemoteService.cpp',
     ]
-- 
2.9.3

