From 3b5ce908da3b7c9fced129d7c96a947ad70b07de Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Wed, 12 Jul 2017 11:25:49 +0200
Subject: [PATCH 6/8] Map Wayland subsurface only when GdkWindow is already
 mapped

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 widget/gtk/mozcontainer.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/widget/gtk/mozcontainer.c b/widget/gtk/mozcontainer.c
index 6f8a4d58a72d..2f51a5fe129b 100644
--- a/widget/gtk/mozcontainer.c
+++ b/widget/gtk/mozcontainer.c
@@ -560,8 +560,15 @@ moz_container_add(GtkContainer *container, GtkWidget *widget)
 struct wl_surface*
 moz_container_get_wl_surface(MozContainer *container)
 {
-    if (!container->subsurface || !container->surface)
+    GdkWindow* window;
+    if (!container->subsurface || !container->surface) {
+        window = gtk_widget_get_window(GTK_WIDGET(container));
+        if (!gdk_window_is_visible(window))
+            return NULL;
+
         moz_container_map_surface(container);
+    }
+
     return container->surface;
 }
 #endif
-- 
2.11.0

