From 32f4f0a3c55d9e8770005f74ba1cf3362bf046d9 Mon Sep 17 00:00:00 2001
From: Martin Stransky <stransky@redhat.com>
Date: Tue, 4 Apr 2017 13:25:47 +0200
Subject: [PATCH 23/44] Improved wayland configure defines

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 toolkit/moz.configure | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 49820b7fe72f..bf149573aa60 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -197,13 +197,17 @@ add_old_configure_assignment('MOZ_X11', x11)
 # Wayland support
 # ==============================================================
 option('--enable-wayland', env='MOZ_WAYLAND',
-       help='Enable Gtk3 Wayland backend build.')
+       help='Enable Gtk3 Wayland backend.')
 
 @depends('--enable-wayland')
 def wayland(value):
-    if value:
+   if value:
         return True
 
+# Check for Wayland headers
+wayland_headers = pkg_check_modules('MOZ_WAYLAND', 'gtk+-wayland-3.0 >= 3.20',
+                                    when=wayland)
+
 set_config('MOZ_WAYLAND', wayland)
 set_define('MOZ_WAYLAND', wayland)
 imply_option('--enable-wayland', wayland)
-- 
2.11.0

