From 1405f4a72144d49329a14558f80adf0d7ddb073f Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 22 Aug 2017 14:54:54 +0900
Subject: [PATCH] Add workarround to reduce unexpected window resize on Wayland

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 widget/gtk/nsWindow.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index a1311f5..14d25c67 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -4330,6 +4330,15 @@ nsWindow::NativeShow(bool aAction)
             if (mWindowType != eWindowType_invisible) {
                 SetUserTimeAndStartupIDForActivatedWindow(mShell);
             }
+
+            // Ensure to resize the window before showing the window.
+            // Resizing GdkWindow is required before resizing GtkWindow.
+            // See the implementation of gtk_window_get_remembered_size() and
+            // gtk_window_compute_configure_request_size() for more detail.
+            GdkRectangle size = DevicePixelsToGdkSizeRoundUp(mBounds.Size());
+            gdk_window_resize(gtk_widget_get_window(mShell), size.width, size.height);
+            gtk_window_resize(GTK_WINDOW(mShell), size.width, size.height);
+
             gtk_widget_show(mShell);
         }
         else if (mContainer) {
-- 
1.9.1

